tests/test_unittest/test_backend.py

```diff
@@ -83,8 +83,9 @@ def test_generate_pre_run(backend_name, mode):
     module_generator.args_data = args_data
     source = module_generator.generate_pre_run()
 
-    if backend_cls.storage_info["device"] == "cpu":
-        assert source == ""
+    if backend_name in CPU_BACKENDS:
+        if backend_name not in DAWN_CPU_BACKENDS:
+            assert source == ""
```

Comment:
This assertion could be simplified to `assert source or backend_name in DAWN_CPU_BACKENDS`

Reply:
:+1:

Reply:
I know it's not pretty, but I formulated the code in this way so to make clear that we need to diversify between GT and Dawn backends.

Reply:
Plus it may serve as a remainder that Dawn backends need more meticulous testing here.